### PR TITLE
Enhancing Job Listing with Filtering

### LINKED ISSUE #16: [feature]Enhancing Job Listing with Filtering
Filter System: Implemented filtering in JobsListView for job types: all, full-time, and part-time.

UI Enhancements: Improved filter button styles with clear active state indication.

Performance: Used useMemo for efficient updates when filters change.

Acceptance Criteria
Filters available for all, full-time, and part-time jobs.
Active filter button is highlighted.
Job list updates only on filter change.

### PR DESCRIPTION
**Pull request by MarthalaSaiKavya**
This PR fixes #16 

Filter Implementation: Added a filtering system in JobsListView to allow users to filter job listings by type (all, full-time, part-time). This feature improves user navigation and helps applicants find relevant job opportunities more efficiently.

UI Improvements: Updated the styling of filter buttons for better visual distinction and user interaction. The selected filter button is highlighted to indicate the active filter.

Code Optimization: Utilized useMemo for efficient filtering, ensuring that the job list updates only when necessary, thus optimizing performance.

### CODE REVIEW

File: frontend/src/components/Job/JobListView.tsx [hunk 1]
```diff
@@ -1,22 +1,57 @@
-import { useEffect } from "react";
+import { useEffect, useState, useMemo } from "react";
 import JobListTile from "./JobListTile";
 
-const JobsListView = (props: any) => {
-  const { jobsList, title } = props;
+interface Job {
+  _id: string;
+  type: string;
+  // Add other job properties as needed
```
Comment (AShivan26) [inline frontend/src/components/Job/JobListView.tsx]: remove the comments

**Review by AShivan26 — CHANGES_REQUESTED** [review]
1.Need documentation in the Readme file
2.Create an Issue in Issue tab and Link it to the PR
3.I see testcases missing
4.The workflow is broken,fix that

File: frontend/src/components/Job/JobListView.tsx (hunk 1 above)
Comment (MarthalaSaiKavya) [reply frontend/src/components/Job/JobListView.tsx]: Sure, removed the comment

**Comment by MarthalaSaiKavya** [comment]
Sure, fixed all of them that you suggested. Thank you for pointing out.

**Comment by MarthalaSaiKavya** [comment]
Sure, made the fixes that you suggested. Thank you for pointing out.

**Review by AShivan26 — APPROVED** [review]

**Pull request merged**